app/cdap/components/Connections/Create/ConnectionConfiguration/ConnectionConfigForm.tsx

```diff
@@ -67,10 +67,7 @@ export function ConnectionConfigForm({
   const onCreate = () => {
     const fieldErrors = {};
     for (const [fieldName, value] of Object.entries(connectorProperties)) {
-      if (
-        !!value.required &&
-        (!values.hasOwnProperty(fieldName) || values[fieldName] === '' || values[fieldName] == null)
-      ) {
+      if ((value as any).required && (!values.hasOwnProperty(fieldName) || !values[fieldName])) {
```

Comment:
The build is reporting error: `Property 'required' does not exist on type 'unknown'.`

Also `values[fieldName] === '' || values[fieldName] == null` is basically equal to `!values[fieldName]`

Reply:
My bad, I missed the case that `values[fieldName] = false` and they don't equal in this corner case. Updated

Reply:
The first error makes sense in the eyes of TS compiler. No guarantee a type `any` has a `required` prop. But for your second statement, I'm guessing we want to avoid treating `values[fieldName] === 0` as `false`

Reply:
You also made a nice counter example, thanks!